### PR TITLE
Service container registrations update

### PR DESCRIPTION
**Pull request by rebornix**
Re #10152 

This is part of refactoring `src/platform` and the PR is mostly moving type registrations from `src/platform` into where they are implemented, i.e, kernel type should be registered in `src/kernels`.

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for feature-requests.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/interactive-window/serviceRegistry.node.ts [hunk 1]
```diff
@@ -63,4 +64,9 @@ export function registerTypes(serviceManager: IServiceManager) {
         undefined,
         [IExtensionSingleActivationService]
     );
+    serviceManager.addSingleton<IJupyterExtensionBanner>(
```
Comment (rebornix) [inline src/interactive-window/serviceRegistry.node.ts:67]: `InteractiveShiftEnterBanner` registered directly in `src/interactive-window`

File: src/kernels/serviceRegistry.web.ts [hunk 2]
```diff
@@ -86,4 +87,9 @@ export function registerTypes(serviceManager: IServiceManager, isDevMode: boolea
     registerJupyterTypes(serviceManager, isDevMode);
 
     serviceManager.addSingleton<CellOutputDisplayIdTracker>(CellOutputDisplayIdTracker, CellOutputDisplayIdTracker);
+
+    // debugging
+    serviceManager.addSingleton<IDebugLocationTracker>(IDebugLocationTracker, DebugLocationTrackerFactory, undefined, [
```
Comment (rebornix) [inline src/kernels/serviceRegistry.web.ts:92]: `DebugLocationTrackerFactory` registered in `src/kernels`.

File: src/notebooks/serviceRegistry.node.ts [hunk 3]
```diff
@@ -89,4 +90,8 @@ export function registerTypes(serviceManager: IServiceManager) {
         IExtensionSingleActivationService,
         EmptyNotebookCellLanguageService
     );
+
+    serviceManager.addSingleton<IDebuggingManager>(IDebuggingManager, DebuggingManager, undefined, [
```
Comment (rebornix) [inline src/notebooks/serviceRegistry.node.ts:94]: `DebuggingManager` registered directly in `src/notebooks`

File: src/webviews/extension-side/serviceRegistry.web.ts [hunk 4]
```diff
@@ -8,17 +8,34 @@ import { WebviewViewProvider } from './webviewViews/webviewViewProvider';
...
 import { DataViewerCommandRegistry } from './dataviewer/dataViewerCommandRegistry';
 import { CommandRegistry as ExportCommandRegistry } from './import-export/commandRegistry';
+import { NotebookWatcher } from './variablesView/notebookWatcher';
+import { DataViewerFactory } from './dataviewer/dataViewerFactory';
+import { ExtensionSideRenderer, IExtensionSideRenderer } from './renderer';
 
 export function registerTypes(serviceManager: IServiceManager, _isDevMode: boolean) {
     serviceManager.add<IWebviewViewProvider>(IWebviewViewProvider, WebviewViewProvider);
     serviceManager.add<IWebviewPanelProvider>(IWebviewPanelProvider, WebviewPanelProvider);
+
```
Comment (rebornix) [inline src/webviews/extension-side/serviceRegistry.web.ts:29]: DataViewer, Varaible Views and renderer registration moved out of `src/platform`

**Review by rchiodo — APPROVED** [review]

**Pull request merged**